src/engine/ruleset/decoders/sonicwall/firewall.yml

```diff
@@ -0,0 +1,72 @@
+---
+name: decoder/sonicwall-firewall/0
+
+metadata:
+  module: Firewall
+  title: Sonicwall-Firewall logs decoder
+  description: Decoder for Firmware module logs belonging to Sonicwall
+  version: [6.5]
+  author:
+    name: Wazuh Inc. info@wazuh.com
+    date: 2023-01-12
+  references:
+    - https://www.sonicwall.com/es-mx/support/technical-documentation
+
+sources:
+  - decoder/syslog/0
+  - decoder/queue-localfile/0
+
+# Since the event is first parsed by the syslog, event.original is
+# not used here but the message field where the syslog saved the rest of the event information.
+check:
+  - message: +r_match/^id\=[[:alpha:]]*? sn\=[[:alnum:]]*? time\=.*? fw
+
+parse:
+  logpar:
+    # The syslog decoder captures this part of the original event "Jan 3 13:45:36 192.168.5.1"
+    # and stores in message this "id=firewall sn=000SERIAL time="2007-01-03 14:48:06" fw= 89.160.20.156 pri=6
+    # c=262144 m=98 msg="Connection Opened" n=23419 src=2.2.2.2:36701:WAN dst=89.160.20.156:50000:WAN proto=tcp/50000"
+
+    # Jan  3 13:45:36 192.168.5.1 id=firewall sn=000SERIAL time="2007-01-03 14:48:06" fw=89.160.20.156 pri=6 c=262144 m=98 msg="Connection Opened" n=23419 src=2.2.2.2:36701:WAN dst=89.160.20.156:50000:WAN proto=tcp/50000
+    - message: <~tmp/kv/=/ /"/'>
+
+normalize:
+  - map:
+      - event.code: $~tmp.m
+      - event.dataset: sonicwall.firewall
+      - event.kind: event
+      - event.module: sonicwall
+      - fileset.name: firewall
+      - input.type: log
+      - observer.product: Firewalls
+      - observer.type: Firewall
+      - observer.vendor: Sonicwall
+      - rsa.internal.messageid: $~tmp.m
+      - rsa.time.event_time: $event.start
+      - service.type: sonicwall
+      - tags: +a_append/forwarded/sonicwall.firewall
+      - \@timestamp: $event.start
+  - check:
+      - ~tmp.src: +ef_exists
+    logpar:
+      - ~tmp.src: <~tmp.src.ip/ip>:<~tmp.src.port/long>:<~tmp.src.interface.name>
+    map:
+      - observer.ingress.interface.name: $~tmp.src.interface.name
+      - rsa.network.sinterface: $~tmp.src.interface.name
+      - source.ip: $~tmp.src.ip
+      - source.port: $~tmp.src.port
+      - related.ip: +a_append/$~tmp.src.ip
+  - map:
+      - ~tmp.proto: +s_trim/end/\/
```

Comment:
This shouldn't work right now, we don't have implemented scaping chars from helpers arguments

Reply:
good catch!